src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -1386,6 +1389,10 @@ private void handlePutBlob(HttpServletRequest request,
                 }
             }
 
+            if (blobStoreType.equals("azureblob")) {
+                eTag = BaseEncoding.base16().lowerCase()
+                        .encode(BaseEncoding.base64().decode(contentMD5String));
+            }
```

Comment:
Let's ignore base16 and base64 encoding for a moment.  The client may provide a Content-MD5 header during a PUT which the server uses to check end-to-end data integrity.  During this operation AWS always returns an ETag which is a MD5 hash of the object.  Clients can use this to ensure the end-to-end integrity of the PUT request.  This pull request removes this feature by returning the *client* Content-MD5 instead of the *server* ETag MD5.  Even if you add a null check for the client Content-MD5, what value will you return for ETag instead?  This pull request also removes support for conditional GET by changing the ETag.

I do not want to discourage work on a higher-fidelity mapping of S3 to Azure, but have you thought through other issues like multi-part upload which returns yet another form of ETag?  Doing a complete emulation of Content-MD5 ETag will be quite involved including mapping the GET, HEAD, and list operations.